File: Basic/Binary representation of next number.cpp
//{ Driver Code Starts
 
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution
{
    public:
    string binaryNextNumber(string S)
    {
        //code here.
        int n = S.length();
        string ans = "";
        int primozero = -1;
        
        for (int i=n-1; i>=0; i--)
        {
            if (S.at(i) == '0')
            {
                primozero=i;
                break;
            }
        }
        
        if (primozero < 0)
        {
            ans += '1';
            for (int i=0; i<n; i++)
            {
                ans +='0';
            }
            return ans;
        }
        
        bool ok = false;
        for (int i=0; i<primozero; i++)
        {
            if (ok)
            {
                ans += S.at(i);
            }
            else
            {
                if (S.at(i) == '1')
                {
                    ok = true;
                    ans += S.at(i);
                }
            }
            
        }
        
        ans += '1';
        
        for (int i=primozero+1; i<n; i++)
        {
            ans += '0';
        }
        

        return ans;
        
    }
};

//{ Driver Code Starts.
int main() 
{
   	int t;
   	cin >> t;
   	while(t--)
   	{
   		string s;
   		cin >> s;
   		Solution ob;
   		cout <<ob.binaryNextNumber(s);
   		cout << "\n";
   	}

    return 0;
}
// } Driver Code Ends

File: Easy/Binary representation of next number.py
#User function Template for python3
class Solution:
	def binaryNextNumber(self, s):
		# code here
		n = len(s)
		
		c = 0
		for c in range(n):
            if s[c] == '1':
                break
        
        s = s[c::]
        
        n = len(s)
		
		ans = ""
		added = False
		
		for i in range(n - 1, -1, -1):
            #print(s[i], added);
            
            if added == True:
                ans = s[i] + ans
            else:
                if s[i] == '0':
                    added = True
                    ans = '1' + ans
                else:
                    ans += '0';
  
        if added == False:
            ans = '1' + ans

        return ans;

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    T = int(input())
    for i in range(T):
        S = input()
        ob = Solution()
        ans = ob.binaryNextNumber(S)
        print(ans)

# } Driver Code Ends
